test/e2e/storage/testsuites/fsgroupchangepolicy.go

```diff
@@ -0,0 +1,278 @@
+		init()
+		defer cleanup()
+
+		TestPodFsGroupChangePolicyAlways(l.config.Framework, l.config.ClientNodeSelection, l.resource.Pvc)
+	})
+
+	ginkgo.It("pod fsgroup change policy (OnRootMismatch) should change the volume permission to match fsgroup as per root directory ownership [LinuxOnly]", func() {
+		init()
+		defer cleanup()
+
+		TestPodFsGroupChangePolicyOnRootMismatch(l.config.Framework, l.config.ClientNodeSelection, l.resource.Pvc)
+	})
+}
+
+// TestPodFsGroupChangePolicyAlways tests 'Always' policy.
+func TestPodFsGroupChangePolicyAlways(f *framework.Framework, nodeSelection e2epod.NodeSelection, pvc *v1.PersistentVolumeClaim) {
+	gid1 := 2000
+	alwaysPodFsGroupPolicy := v1.PodFSGroupChangePolicy("Always")
+	podConfig := e2epod.Config{
+		NS:                     f.Namespace.Name,
+		NodeSelection:          nodeSelection,
+		PVCs:                   []*v1.PersistentVolumeClaim{pvc},
+		FsGroup:                utilpointer.Int64Ptr(int64(gid1)),
+		PodFSGroupChangePolicy: &alwaysPodFsGroupPolicy,
+	}
+	ginkgo.By(fmt.Sprintf("Creating Pod in namespace %s with fsgroup %s", f.Namespace.Name, strconv.Itoa(gid1)))
+	pod, err := e2epod.CreateSecPodWithNodeSelection(f.ClientSet, &podConfig, framework.PodStartTimeout)
+	framework.ExpectNoError(err)
+	framework.Logf("First Pod %s/%s started successfully", pod.Namespace, pod.Name)
+	cmd := "ls -l /mnt/volume1"
+	stdout, stderr, err := storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+	framework.Logf("pod exec for cmd %s, stdout: %s, stderr: %s", cmd, stdout, stderr)
+
+	ginkgo.By(fmt.Sprintf("Creating a sub-directory and file, and verifying their ownership is %s", strconv.Itoa(gid1)))
+	cmd = "touch /mnt/volume1/file1"
+	_, _, err = storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+	storageutils.VerifyFilePathGidInPod(f, "/mnt/volume1/file1", strconv.Itoa(gid1), pod)
+
+	cmd = "mkdir /mnt/volume1/subdir"
+	_, _, err = storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+	cmd = "touch /mnt/volume1/subdir/file2"
+	_, _, err = storageutils.PodExec(f, pod, cmd)
+	framework.ExpectNoError(err)
+	storageutils.VerifyFilePathGidInPod(f, "/mnt/volume1/subdir/file2", strconv.Itoa(gid1), pod)
+
+	ginkgo.By(fmt.Sprintf("Deleting Pod %s/%s", pod.Namespace, pod.Name))
+	framework.ExpectNoError(e2epod.DeletePodWithWait(f.ClientSet, pod))
+
+	gid2 := 3000
+	podConfig.FsGroup = utilpointer.Int64Ptr(int64(gid2))
+	pod = createPodAndVerifyContentGid(f, &podConfig)
+
+	gid3 := 4000
+	ginkgo.By(fmt.Sprintf("Changing the sub-directory file ownership to %s", strconv.Itoa(gid3)))
+	storageutils.ChangeFilePathGidInPod(f, "/mnt/volume1/subdir/file2", strconv.Itoa(gid3), pod)
+
+	ginkgo.By(fmt.Sprintf("Deleting Pod %s/%s", pod.Namespace, pod.Name))
+	framework.ExpectNoError(e2epod.DeletePodWithWait(f.ClientSet, pod))
+
+	podConfig.FsGroup = utilpointer.Int64Ptr(int64(gid2))
```

Comment:
By("Verifying that the ownership gets changed back to gid 3000 when the pod restarts"